lib/engine/game/g_1848/step/buy_sell_par_shares.rb

```diff
@@ -15,6 +15,18 @@ def process_buy_shares(action)
                        allow_president_change: @game.pres_change_ok?(action.bundle.corporation))
             track_action(action, action.bundle.corporation)
           end
+
+          def can_sell?(entity, bundle)
+            return true if bundle.corporation == @game.boe
+
+            super
+          end
+
+          def can_buy?(entity, bundle)
+            return true if bundle.corporation == @game.boe
```

Comment:
Does this handle not going over the cert limit?

Reply:
can_buy? doesn't handle cert limit even in the parent method, it's handled in must_sell?

Reply:
can_buy? calls can_gain?, which checks the cert limit. You'll need to check cert limit here.

Reply:
i removed the overridden method.
It's not clear from the rules wether you can or can't buy a share if you've already sold in the same round.
I will keep the behaviour the same as other 18xx games.

Reply:
> **12.3.2** If a player has sold shares in a company in a SR, he may not buy shares in the same company during this same SR.

So it's the same as the usual.

If you're talking about the Bank of England, then at least in the original rules, it's specified that it counts as a company except for explicit exceptions.

Reply:
ok, so my understanding is correct. I was working off a checklist someone else made, where it said no restrictions on boe shares at all. Current code is accurate with the rules